### ISSUE TITLE
Upbeat Giraffe bugs

### ISSUE THREAD

**Issue by wizzomafizzo** [wizzomafizzo/starcheat#134]
This is a master ticket for known issues coming from the latest Starbound/starcheat update.

- Equipping techs corrupts players #132 

Easy to reproduce, just equip a tech on a tech slot. Learning/unlearning techs seems to work fine.

It appears Starbound also attempts to do some cleaning when it tries to load a bad tech player which causes starcheat to have a mini crash when they're cleared from the slots later.

Need to take a look at a player file with legit equipped techs and see what is missing exactly. Also find out if there are any techs than can be equipped fine through starcheat.

- Some items crash starcheat when viewed #120 
- Issue generating new swords #133 
- Unconfirmed issue indexing mods #126 

Still waiting on a response but I have a wild hunch this could be caused by Starbound allowing old packed mods (v2) to load up along with new (v3).

- Issue changing player appearance #124 

I still can not reproduce this crash. Need an example corrupt player that doesn't also have corrupted techs.

- Old json exports not compatible #123

Not sure if this can be "fixed" in starcheat itself so much without major work

**Comment by MavJones** [comment]
I believe #124  is probably just a coincidence of #132 that was said be be a crash due to appearance.

**Comment by Broswirl** [comment]
#132 Is a Crash of the Playerfile and Starcheat, Starcheat is not able to remove the Tech again (error massage) and Starbound can´t load the Playerfile (missing in the list).

**Comment by wizzomafizzo** [comment]
i think tech issues should be fixed and recoverable now in dev

**Comment by wizzomafizzo** [comment]
@Ristovski are you able to reproduce this reliably? it is really weird that it would do that, it shouldn't be editing that part of the save at all

**Comment by wizzomafizzo** [comment]
something to think about later, these 2 errors appeared for most issues, maybe they should be handled more gracefully (missing icon image, log error). from what i can tell even though they obviously point to a greater problem, they shouldn't technically be critical enough to kill the whole app

```
Traceback (most recent call last):
File "C:\Python34\lib\site-packages\cx_Freeze\initscripts\Console.py", line 27, in 
File "build\starcheat.py", line 74, in 
File "build\starcheat.py", line 71, in main
File "build\gui\mainwindow.py", line 134, in init
File "build\gui\mainwindow.py", line 426, in open_file
File "build\gui\mainwindow.py", line 195, in update
File "build\gui\mainwindow.py", line 554, in update_bag

KeyError: '__content'
```

```
Traceback (most recent call last):
File "C:\Python34-x64\lib\site-packages\cx_Freeze\initscripts\Console.py", line 27, in 
File "build\starcheat.py", line 74, in 
File "build\starcheat.py", line 71, in main
File "build\gui\mainwindow.py", line 134, in init
File "build\gui\mainwindow.py", line 417, in open_file
File "build\gui\utils.py", line 332, in init
File "build\gui\utils.py", line 375, in populate
File "build\assets.py", line 1093, in render_player
TypeError: 'NoneType' object is not subscriptable
```

**Comment by wizzomafizzo** [comment]
intermittent issue i'm having trouble reproducing. sometimes starcheat forget the player folder location configured and looks in the old root player folder instead. it will immediately offer to reconfigure and reindex, then it works fine again until n number of restarts

**Comment by chrmoritz** [comment]
@wizzomafizzo I thought I've fixed this in 8376052.

**Comment by wizzomafizzo** [comment]
that was part of it but in the options dialog on save it would regen the paths using the old method and set them directly in the config. i was able to reproduce it every time by just clicking save from the options dialog

it always sorted itself out but it should be fixed in the latest commit anyway

**Comment by chrmoritz** [comment]
Yeah, I see. When do you plan to release a new version with the fixes?

**Comment by wizzomafizzo** [comment]
lol i was waiting for you to do it since you did the other hotfixes. i am happy to release it whenever, i think we've got all the serious bugs worked out now

i can do it but i'm at work so not for at least another 6 hours. if you wanna do it before then go ahead though

**Comment by Ristovski** [comment]
@wizzomafizzo Can't really check right now since I just installed a brand new system, will test when I get a chance. Do you plan on adding support for a "Master Unlocker" option in StarCheat that will unlock all quests/techs?

**Comment by wizzomafizzo** [comment]
@Ristovski i can't say exactly when since it's pretty low priority but absolutely quest management is on the list and so is that magic unlock button

**Comment by FoggyPaws** [comment]
I am really hoping that the next version lets me put ?replace data after image directories
cause the "image" json thingamajig doesn't like when I have anything but an image directory in there, adding ?replace stuff makes it explode :(

**Comment by wizzomafizzo** [comment]
will do!

**Comment by AuroraDestiny** [comment]
I'm still having issues generating Broadswords, which are useful templates for making drawables. Is there an update on when that may be "repaired" ?

Thanks in advance.

**Comment by chrmoritz** [comment]
You should open an issue if all the details if there is a problem wit it.

**Comment by wizzomafizzo** [comment]
Ya definitely sounds the same as #133 
It is pretty high on the list, I also need to look at shields, saplings and capture pods

If you have a character with legit filled capture pods and shield please upload it!

**Comment by wizzomafizzo** [comment]
@AuroraDestiny this is fixed in the dev branch now, will be in the next release

**Comment by wizzomafizzo** [comment]
remaining issues
- [ ] sapling generation is completely broken, corrupts save
- [ ] capture pod generation produces valid items but no monster (pretty sure it used to do that?)
- [ ] shield generation works but apparently the recoil key needs to be changed
- [ ] handle missing images/__content, https://github.com/wizzomafizzo/starcheat/issues/134#issuecomment-72854150
- [ ] tech module names need to be indexed and used on equip
- [ ] this mod bug #157 #137

**Comment by wizzomafizzo** [comment]
everything here is fixed as of the latest commits to dev. look out for the next release (shouldn't be too much longer)
